### DISCUSSION TITLE
NanoELS separate box for control panel, Pictures

### DISCUSSION THREAD

**Discussion by wberggren** [kachurovskiy/nanoels#13] [Show and tell]
First of all many thanks to Maxim for his excellent work and very good description. I have an Optimum lathe which has a smaller diameter spindle. This forced me to select a larger gear wheel (100 teeth) for the encoder. The encoder that I had was a 1024 pulses Omron 26B2 encoder from ebay. The speed can be easily corrected in the software. I also used a servo stepper motor which works fine. Here are the pictures:

![IMG_7739](https://user-images.githubusercontent.com/80155851/124111608-c404c400-da69-11eb-9582-49f27edbedd5.jpg)

![IMG_7742](https://user-images.githubusercontent.com/80155851/124111834-fca49d80-da69-11eb-9545-1b731acee2ed.jpg)

![IMG_7744](https://user-images.githubusercontent.com/80155851/124111670-d41ca380-da69-11eb-984c-0da8f1c5f8aa.jpg)

![IMG_7743](https://user-images.githubusercontent.com/80155851/124111694-daab1b00-da69-11eb-9842-194b20b7b91b.jpg)

![IMG_7742](https://user-images.githubusercontent.com/80155851/124111725-e3035600-da69-11eb-8c9f-051163a5369c.jpg)

![IMG_7741](https://user-images.githubusercontent.com/80155851/124111853-03331500-da6a-11eb-8b3d-8c326c9f1940.jpg)

![IMG_7740](https://user-images.githubusercontent.com/80155851/124111867-075f3280-da6a-11eb-858d-6ed1a3b17601.jpg)

![IMG_7739](https://user-images.githubusercontent.com/80155851/124111943-1ba32f80-da6a-11eb-9771-408b874411c5.jpg)

**Comment by kachurovskiy** [comment]
Such an awesome build, thanks for sharing! Very well done.

**Reply by RGUERI1** [reply]
What setting needs to be modified, if any, if you are using a larger gear for the encoder? I'm using a v-belt drive because I couldn't squeeze the gear behind the pulleys, so I'm no longer at a 1:1 ratio. However, I didn't see a setting in the code to modify.

**Reply by kachurovskiy** [reply]
@RGUERI1 calculate the number of pulses encoder sends per revolution of the spindle and use it in `ENCODER_STEPS` if you're building H2 - https://github.com/kachurovskiy/nanoels/blob/main/h2/h2.ino#L6 - cheers!

**Comment by RGUERI1** [comment]
Thanks for speedy reply. I have the Nano 4 but I saw that parameter in the ino file. Assume it’s the same function?
Thanks,
Rustin

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by kachurovskiy** [reply]
Yes, it's same parameter in the same line - https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L6 - thanks!

**Comment by RGUERI1** [comment]
Ok, my encoder pulley is smaller than the spindle pulley that I'm using. I
come up with 1.644 turns of the encoder for every 1-turn of the spindle. If
my math is correct that is 986.664 pulses per spindle rev.
Does that make sense?

**Reply by kachurovskiy** [reply]
Perhaps we should take this to a new discussion and keep this thread dedicated to the H2 build above. Sadly fractional PPR is not supported, we rely on it being convertible to integer pretty hard.

**Comment by RGUERI1** [comment]
Oops, I thought this was just an email between us! . So in summary it needs to come out to a whole number to work correctly?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
